From e8edb6746fee96f15610027c49e4d67756092aac Mon Sep 17 00:00:00 2001
From: Ayrton <a.munoz3327@gmail.com>
Date: Thu, 6 Jun 2024 10:59:16 -0400
Subject: [PATCH] Makefile: Fix build when using clang as AS

Unlike gcc, clang enforces -mgeneral-regs-only when used as the assembler so
this removes that flag to fix the error in #393. This is only needed for
utils_asm.S which has code saving/restoring SIMD state, but this commit applies
it to all asm files for simplicity since code for fp regs shouldn't be generated
unless they're explicitly used anyway.

Signed-off-by: Ayrton Munoz <a.munoz3327@gmail.com>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 2620e720..01a1e249 100644
--- a/Makefile
+++ b/Makefile
@@ -193,7 +193,7 @@ build/%.o: src/%.S
 	$(QUIET)echo "  AS    $@"
 	$(QUIET)mkdir -p $(DEPDIR)
 	$(QUIET)mkdir -p "$(dir $@)"
-	$(QUIET)$(AS) -c $(CFLAGS) -MMD -MF $(DEPDIR)/$(*F).d -MQ "$@" -MP -o $@ $<
+	$(QUIET)$(AS) -c $(BASE_CFLAGS) -MMD -MF $(DEPDIR)/$(*F).d -MQ "$@" -MP -o $@ $<
 
 $(BUILD_FP_OBJS): build/%.o: src/%.c
 	$(QUIET)echo "  CC FP $@"
